### PR TITLE
Add master lining action

### PR DESCRIPTION
**Pull request by paddyroddy**
This can be used in other repos (and, in this one!)

Inspiration from https://github.com/UCL/composite-actions/blob/main/.github/workflows/test-python-tox.yml

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -12,21 +12,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout source
-        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4
+        uses: actions/checkout@v4
```
Comment (paddyroddy) [inline .github/workflows/linting.yml:15]: Seems that you require to `checkout` for the action locally, but not when used externally.

See David's example here https://github.com/UCL/composite-actions/blob/aa7d881cf40a6a34a3b93fd619dc2ac7685f9a25/.github/workflows/test-python-tox.yml#L13-L20 vs https://github.com/UCL/composite-actions/blob/aa7d881cf40a6a34a3b93fd619dc2ac7685f9a25/README.md?plain=1#L13-L20

File: actions/linting/action.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.x"
+
+    - name: Install dependencies
+      shell: bash
+      run: python -m pip install pre-commit
+
+    - name: Run pre-commit
+      shell: bash
```
Comment (paddyroddy) [inline actions/linting/action.yml:31]: Looks like `shell` is required for `composite actions` https://stackoverflow.com/a/71042698/7359333

File: actions/linting/action.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  pre-commit-config:
+    description: The pre-commit config file of the current repo
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout source
+      uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4
```
Comment (drmatthews) [inline actions/linting/action.yml:17]:
```suggestion
      uses: actions/checkout@v4
```

File: actions/linting/action.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    - name: Install ansible
+      shell: bash
+      run: python -m pip install ansible
+      if: ${{ inputs.ansible-roles-config != 'missing' }}
+
+    - name: Install pre-commit
+      shell: bash
+      run: python -m pip install pre-commit
+
+    - name: install ansible roles
```
Comment (drmatthews) [inline actions/linting/action.yml:52]:
```suggestion
    - name: Install ansible roles
```

**Review by drmatthews — APPROVED** [review]
👍

File: actions/linting/action.yml (hunk 3 above)
Comment (paddyroddy) [reply actions/linting/action.yml:17]: This is actually suggested my `renovate` for security reasons - think it did it automatically

File: actions/linting/action.yml (hunk 4 above)
Comment (paddyroddy) [reply actions/linting/action.yml:52]: Good spot

**Pull request merged**